src/game/path.service.js

```diff
@@ -58,28 +58,39 @@ angular.module('ngGo.Game.Path.Service', [
       return;
     }
 
+    //Decrement move
+    this.move--;
+
     //Delete path choice
-    if (this.path[this.move]) {
+    if (typeof this.path[this.move] !== 'undefined') {
+      var branch = this.path[this.move];
       delete this.path[this.move];
-      this.branches--;
+      if (branch > 0) {
+        this.branches--;
+      }
     }
 
-    //Decrement move
-    this.move--;
     return this;
```

Comment:
The old code will leave some unnecessary and incorrect data behind. If `this.path[this.move]` is `0`, JavaScript would evaluate it as `false`. And `this.branches` should decrement only when `this.path[this.move]` is non-zero (meaning that it remembered branch line).